core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorAdapterTest.scala

```diff
@@ -436,4 +440,74 @@ class GroupCoordinatorAdapterTest {
 
     assertEquals(expectedResults, future.get())
   }
+
+  @ParameterizedTest
+  @ApiKeyVersionsSource(apiKey = ApiKeys.OFFSET_COMMIT)
+  def testCommitOffsets(version: Short): Unit = {
+    val groupCoordinator = mock(classOf[GroupCoordinator])
+    val time = new MockTime()
+    val adapter = new GroupCoordinatorAdapter(groupCoordinator, time)
+    val now = time.milliseconds()
+
+    val ctx = makeContext(ApiKeys.OFFSET_COMMIT, version)
+    val data = new OffsetCommitRequestData()
+      .setGroupId("group")
+      .setMemberId("member")
+      .setGenerationId(10)
+      .setRetentionTimeMs(1000)
```

Comment:
can we test values less than -1?

Reply:
We could but it does not add any value as the value is just copied to OffsetAndMetadata. As explained earlier, there is not validation for this.